src/main/java/SpringWeatherForecast/DataSource.java

```diff
@@ -0,0 +1,40 @@
+package SpringWeatherForecast;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Optional;
+
+public class DataSource {
```

Comment:
Usually such classes are named as Gateways, for instance, MetaweatherGateway. It is a gateway to the remote service.